include/ensmallen_bits/sgd/sgd_impl.hpp

```diff
@@ -116,6 +116,11 @@ double SGD<UpdatePolicyType, DecayPolicyType>::Optimize(
         std::min(batchSize, actualMaxIterations - i),
         numFunctions - currentFunction);
 
+    if( effectiveBatchSize == 0 )
+	{
+	  break;
+	}
```

Comment:
If effectiveBatchSize is 0 then we should break from this loop otherwise we will be trapped in infinite looping.